thewalrus/fock_gradients.py

```diff
@@ -84,20 +120,16 @@ def Dgate(r, theta, cutoff, grad=False):
     Returns:
         tuple[array[complex], array[complex], array[complex]]: The Fock representations of the gate and its gradients with sizes ``[cutoff]*2``
     """
-    phase = np.exp(1j * theta)
-    y = np.array([r * phase, -r * np.conj(phase)])
-
     if not grad:
-        return displacement(y, cutoff), None, None
-
-    T = displacement(y, cutoff + 1)
+        return displacement_rec(r * np.exp(1j * theta), cutoff), None, None
+    T = displacement_rec(r * np.exp(1j * theta), cutoff + 1)
     gradTr = np.zeros([cutoff, cutoff], dtype=complex)
     gradTtheta = np.zeros([cutoff, cutoff], dtype=complex)
     grad_Dgate(T, gradTr, gradTtheta, theta)
-    return T[0:cutoff, 0:cutoff], gradTr, gradTtheta
+    return T[:cutoff, :cutoff], gradTr, gradTtheta
 
 
-@jit("void(complex128[:,:], complex128[:,:], complex128[:,:], double)")
+@jit(nopython=True)
```

Comment:
How come the Numba dtypes don't have to be specified anymore?

Reply:
Actually you don't need specify them. That was my mistake. I will homogenize it